3rd_party/3rd_party.cmake

```diff
@@ -38,11 +38,11 @@ set(ARCH ${HOST_SYSTEM_PROCESSOR})
 set(CMAKE_SYSTEM_NAME ${HOST_SYSTEM_NAME})

```

Comment:
Why do we change the value of a built in CMake variable?  Can't we just use `${HOST_SYSTEM_NAME}` in the code below instead of using a built in variable with a changed value?

Reply:
Or could we just use `CMAKE_HOST_SYSTEM_NAME` throughout the code below so that we're using a built in variable that people won't have to check the meaning of?